tooling/greybox_fuzzer/src/corpus.rs

```diff
@@ -0,0 +1,431 @@
+
+        file.write_all(contents.as_bytes())
+            .map_err(|_x| format!("Couldn't write to file {:?}", &full_file_path))?;
+        self.file_manager.add_file_with_source(&full_file_path, String::from(contents));
+        Ok(())
+    }
+}
+
+/// Tracks information about a sequence of executions for a particular testcase,
+/// including how many executions remain and which testcase is being used
+#[derive(Debug)]
+struct Sequence {
+    // ID of the testcase being executed in sequence
+    testcase_id: TestCaseId,
+    // Number of remaining executions in this sequence
+    executions_left: u64,
+}
+
+impl Sequence {
+    /// Creates a new empty sequence
+    pub fn new() -> Self {
+        Self { testcase_id: 0, executions_left: 0 }
+    }
+
+    /// Returns true if there are no executions remaining in this sequence
+    pub fn is_empty(&self) -> bool {
+        self.executions_left == 0
+    }
+
+    /// Resets the sequence by setting remaining executions to 0
+    pub fn clear(&mut self) {
+        self.executions_left = 0
+    }
+
+    /// Decrements the number of remaining executions by 1
+    pub fn decrement(&mut self) {
+        self.executions_left -= 1
+    }
+}
+
+/// Manages the selection of testcases for fuzzing, tracking execution counts and scheduling
+pub struct TestCaseOrchestrator {
+    // Maps testcase IDs to their total execution count
+    executions_per_testcase: HashMap<TestCaseId, u64>,
+    // Maps testcase IDs to their sequence number (how many times they've been selected for sequential fuzzing)
+    sequence_number: HashMap<TestCaseId, u32>,
+    // Total number of executions across all testcases
+    total_executions: u64,
+    // Currently active sequence of executions
+    current_sequence: Sequence,
+}
+
+/// Represents the next testcase(s) selected for fuzzing:
+/// - First element is the primary testcase ID
+/// - Second optional element is an additional testcase ID for splicing mutations
+type NextSelection = (TestCaseId, Option<TestCaseId>);
+
+impl TestCaseOrchestrator {
+    /// Creates a new empty TestCaseOrchestrator
+    pub fn new() -> Self {
+        Self {
+            executions_per_testcase: HashMap::new(),
+            sequence_number: HashMap::new(),
+            total_executions: 0,
+            current_sequence: Sequence::new(),
+        }
+    }
+
+    /// Adds a new testcase to be managed by the orchestrator
+    pub fn new_testcase(&mut self, testcase_id: TestCaseId) {
+        self.executions_per_testcase.insert(testcase_id, 0);
+        self.sequence_number.insert(testcase_id, 0);
+    }
+
+    /// Removes a testcase from being managed by the orchestrator.
+    /// Updates execution counts and clears current sequence if it was using this testcase.
+    pub fn remove(&mut self, testcase_id: TestCaseId) {
+        let executions = self.executions_per_testcase[&testcase_id];
+        self.executions_per_testcase.remove(&testcase_id);
```

Comment:
```suggestion
        let executions = self.executions_per_testcase.remove(&testcase_id).unwrap();
```

or `expect` instead of `unwrap`